llmfoundry/registry.py

```diff
@@ -206,6 +206,17 @@
     description=_metrics_description,
 )
 
+_icl_datasets_description = (
+    'The ICL dataloaders registry is used to register an torch.utils.data.Dataset class which can be used for ICL tasks.'
+)
+icl_datasets = create_registry(
+    'llmfoundry',
+    'icl_datasets',
+    generic_type=Type[Dataset],
```

Comment:
For your customization did you want to make this a generic `Dataset`? Otherwise I would suggest having it be `Type[InContextLearningDataset]`

Reply:
I do have to make my own variant of `InContextLearningDataset` which can read from parquet files and is not limited to json file inputs, but yeah I can just inherit from `InContextLearningDataset`.

Reply:
Yeah, I think inheriting makes sense here. We want each registry to have a defined interface, and this one is that it matches the `InContextLearningDataset` interface.

Reply:
One problem I am running into is that the since the `InContextLearningDataset` class is defined in the same place as the builder methods (`build_icl_dataloader`), using `InContextLearningDataset` creates a circular import between this and the builder method. Do you think the refactoring can be taken up as a follow up?

Reply:
Yeah, I'm ok with that. Please note in a comment about this (and the registry description)